bazel/py_proto_library.bzl

```diff
@@ -84,16 +93,21 @@ def _py_proto_aspect_impl(target, ctx):
         else:
             proto_root = ctx.workspace_name + "/" + proto_root
 
+        additional_args = ctx.actions.args()
+        if generated_stubs:
+            additional_args.add(plugin_output, format="--pyi_out=%s")
+
         proto_common.compile(
             actions = ctx.actions,
             proto_info = proto_info,
             proto_lang_toolchain_info = proto_lang_toolchain_info,
-            generated_files = generated_sources,
+            generated_files = generated_sources + generated_stubs,
             plugin_output = plugin_output,
+            additional_args = additional_args,
         )
 
     # Generated sources == Python sources
-    python_sources = generated_sources
+    python_sources = generated_sources + generated_stubs
```

Comment:
We discussed this with some python folks, and while we agree this PR looks like a good approach, it needs a few tweaks.  Specifically, the pyi files should end up in the new `direct_pyi_files` and `transitive_pyi_files` fields of `PyInfo` added in rules_python 1.1, instead of getting globbed together into `transitive_sources` with the py files like is currently happening

I think this is just a matter of piping the information through _PyProtoInfo, and should be a relatively minor tweak to your PR though

Reply:
@mkruskal-google Thank you for your feedback.  At present, it looks like protobuf requires rules_python 1.0.0.  Are you asking for me to bump that to rules_python 1.1.0 as part of this PR or possibly a separate PR?  Or should I attempt to be compatible with rules_python 1.0.0 but take advantage of the new `PyInfo` fields if they are available?

Reply:
Those are the two options :). I think bumping to 1.1 should be fine as it's a minor version release, and it would be a lot easier.  We could go the extra mile to support 1.0.0 by checking for the presence of the new fields, but I don't think it's necessary here

Reply:
@mkruskal-google I've put up https://github.com/protocolbuffers/protobuf/pull/22079 to bump the minimum `rules_python` version to 1.1.0.

Reply:
This lgtm as a starting point.  If grpc requires further tweaks we can make those separately I think